engines/suppliers/app/controllers/suppliers/api/v1/producers_controller.rb

```diff
@@ -0,0 +1,52 @@
+module Suppliers
+  module Api
+    module V1
+      class ProducersController < ApplicationController
+        before_action :authenticate
+        before_action :load_group,
+          only: [:index],
+          if: -> { !!params.fetch('group_id', nil) }
```

Comment:
When there are so many exceptions, IMO there's no point on having it in a before_action and it's way better to explicitly call `load_group` from the index action; easier to read.